hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -4764,29 +4771,25 @@ public String addSqlHintOrComment(
 	}
 
 	/**
-	 * Adds an INDEX query hint as follows:
+	 * Adds an {@code INDEX} query hint as follows:
 	 *
 	 * <pre>
 	 * SELECT *
 	 * FROM TEST
 	 * USE INDEX (hint1, hint2)
 	 * WHERE X=1
 	 * </pre>
+	 *
+	 * @since 7.0
 	 */
-	public static String addQueryHints(String query, String hints) {
-		Matcher matcher = QUERY_PATTERN.matcher( query );
+	public static String addUseIndexQueryHint(String query, String hints) {
```

Comment:
I don't know if we consider `Dialect` to be an API or SPI, but maybe re-add the old method and mark it as deprecated?

```java
@Deprecated(forRemoval = true)
public static String addQueryHints(String query, String hints) {
    return addUseIndexQueryHint( query, hints );
}
```

Reply:
> I don't know if we consider `Dialect` to be an API or SPI

- It's not used to write *application logic* (it's not exposed via `Session` or `SessionFactory`) and therefore it cannot be an API.
- It *is* an interface used to customize or extend Hibernate, and therefore an SPI.

> maybe re-add the old method and mark it as deprecated?

This method was added in 7.0.0.Beta2 so that's not necessary. We don't want to fill the codebase up with dead/useless code.